### DISCUSSION TITLE
What is the cause of that problem: terminate called after throwing an instance of 'std::length_error' what(): cannot create std::vector larger than max_size()

### DISCUSSION THREAD

**Discussion by KirollosMedhat** [casadi/casadi#3117] [Q&A]
Writing a code in casadi-c++ for an MPC(model predictive control)
and i get that runtime error:

terminate called after throwing an instance of 'std::length_error'
what(): cannot create std::vector larger than max_size()

here's the code:

https://gist.github.com/KirollosMedhat/049bea182a7c8eaa34baa1513df059da

**Comment by jaeandersson** [comment]
Not enough information. If it's in CasADi, you need to at least get the stack trace, e.g. from a debugger like gdb or valgrind

**Reply by KirollosMedhat** [reply]
@jaeandersson 
Hello, here's the trace you asked for sorry for being late
please check and tell me what can i do, thanks

[Backtrace.txt](https://github.com/casadi/casadi/files/11311354/Backtrace.txt)

**Comment by jaeandersson** [comment]
My guess is that it has something to do with `double N`. You should use an integer type for dimension. I would have expected a compilation error however.

**Reply by KirollosMedhat** [reply]
I changed it to int 

And here's the new code:
https://gist.github.com/KirollosMedhat/4c3839b0d7dee24e491653c32cd85a3b

**Comment by jaeandersson** [comment]
Your trace doesn't appear to have debug symbols, like the line numbers where the exception happened. Did you compile CasADi in debug mode?

**Reply by KirollosMedhat** [reply]
i rebuilt the project now using the following line in the cmakelists.txt file:
set(CMAKE_BUILD_TYPE Debug)
and i made new stack trace but it is the same exactly.

**Reply by KirollosMedhat** [reply]
I sent my code to my colleague in the graduation project  and the code ran on his machine without problems and the solver gave results.
So apparently the problem is in my machine.

**Comment by jgillis** [comment]
Code runs fine on my machine, too.
What compiler version and CasADi version?

If the stack trace is identical in Debug mode, it indicates something went wrong with producing or using the new CasADi binary.
It should have line numbers added.

**Comment by MaxxWilson** [comment]
I had this exact issue just yesterday using the main branch and attempting to run the rocket_ipopt example. I am using Ubuntu 20.04.

**To solve**, I updated from GCC/G++ 9 to 11 as described [here](https://github.com/zehuilu/Tutorial-on-CasADi-with-CPP) and I built IPOPT and MUMPS from source as opposed to using the available binaries (coinor-libipopt1v5-dbg/focal 3.11.9-2.2build2 was the available package at time of writing).

Not sure if this will fix the issue for others, but figured it would be good to leave a note as this page was the only relevant result I got when searching the original error code.

Cheers!
